src/Package/Impl/ProjectSystem/ProjectSystemServices.cs

```diff
@@ -28,7 +29,11 @@ internal sealed class ProjectSystemServices : IProjectSystemServices {
             DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
             if (dte.Solution.Projects.Count > 0) {
                 try {
-                    return dte.Solution?.Projects?.Cast<EnvDTE.Project>()?.First();
+                    var projects = dte.Solution?.SolutionBuild?.StartupProjects as IEnumerable;
```

Comment:
If it's actually `IEnumerable` (which your change implies by casting it to one), then you can do `Cast` to get a typed enumerable out of it.